File: adventofcode-2016/src/test/java/nl/mvdr/adventofcode/adventofcode2016/day12/Part1Test.java
package nl.mvdr.adventofcode.adventofcode2016.day12;

import java.util.stream.Stream;

import org.junit.jupiter.params.provider.Arguments;

import nl.mvdr.adventofcode.solver.SolverTest;

class Part1Test extends SolverTest<Part1> {

    public Part1Test() {
        super(Part1.class);
    }
    
    static Stream<Arguments> testSolution() {
        return Stream.of(Arguments.of("42", "example-day12.txt"));
    }
}

File: adventofcode-2025/src/test/kotlin/nl/mvdr/adventofcode/adventofcode2025/day06/Part1Test.kt
package nl.mvdr.adventofcode.adventofcode2025.day06

import nl.mvdr.adventofcode.solver.FunctionSolverTest
import org.junit.jupiter.params.provider.Arguments

class Part1Test: FunctionSolverTest<Long>(::solvePart1) {
    companion object {
        @JvmStatic
        fun testSolution(): List<Arguments> {
            return listOf(
                Arguments.of("4277556", "example-day06.txt")
            )
        }
    }
}